### PR TITLE
update sceneViewsettings

### PR DESCRIPTION
**Pull request by Barsonax**

### CODE REVIEW

**Comment by Barsonax** [comment]
Ready for a review @ilexp @SirePi

File: Source/Plugins/EditorModules/SceneView/SceneViewSettings.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿namespace Duality.Editor.Plugins.SceneView
+{
+	public class SceneViewSettings
+	{
+		private bool showComponents = true;
+
+		public bool ShowComponents
```
Comment (ilexp) [inline Source/Plugins/EditorModules/SceneView/SceneViewSettings.cs:7]: Class and property need an XML comment. (The property controls whether components are shown as individual nodes in the scene tree.)

File: Source/Plugins/EditorModules/SceneView/Modules/SceneView.cs [hunk 2]
```diff
@@ -77,6 +74,14 @@ private class CreateContextEntryTag
...
 		private MenuModelItem	nodeContextItemPaste	= null;
 
+		private SceneViewSettings userSettings = new SceneViewSettings();
+
+		public SceneViewSettings UserSettings
+		{
+			get { return this.userSettings; }
+			set { this.userSettings = value; }
+		}
+
```
Comment (ilexp) [inline Source/Plugins/EditorModules/SceneView/Modules/SceneView.cs:84]: Style consistency: Remove the newline between properties to group them closely.

File: Source/Plugins/EditorModules/SceneView/Modules/SceneView.cs [hunk 3]
```diff
@@ -77,6 +74,14 @@ private class CreateContextEntryTag
 		private MenuModelItem	nodeContextItemCopy		= null;
 		private MenuModelItem	nodeContextItemPaste	= null;
 
+		private SceneViewSettings userSettings = new SceneViewSettings();
```
Comment (ilexp) [inline Source/Plugins/EditorModules/SceneView/Modules/SceneView.cs:77]: Style consistency: Use a double newline to separate members of different kinds if either of them is grouped into semantic blocks on their own.

(This was probably wrong before your change, but might as well address it while here)

File: Source/Plugins/EditorModules/SceneView/Modules/SceneView.cs [hunk 4]
```diff
@@ -171,17 +176,10 @@ protected override void OnClosed(EventArgs e)
 			Scene.ComponentAdded -= this.Scene_ComponentAdded;
 			Scene.ComponentRemoving -= this.Scene_ComponentRemoving;
 		}
-
-		internal void SaveUserData(XElement node)
-		{
-			node.SetElementValue("ShowComponents", this.buttonShowComponents.Checked);
-		}
-		internal void LoadUserData(XElement node)
+
```
Comment (ilexp) [inline Source/Plugins/EditorModules/SceneView/Modules/SceneView.cs:179]: Style consistency: Another drive-by fix, this one should be a double newline as well.
